src/test/java/com/aws/greengrass/authorization/WildcardTrieTest.java

```diff
@@ -16,51 +16,111 @@
 class WildcardTrieTest {
     @Test
     void testGlobWildcardMatching() {
+        // no wildcards
         WildcardTrie rt = new WildcardTrie();
-        rt.add("abc*xyz*");
-        assertTrue(rt.matchesMQTT("abc1234asdxyz456" ));
-        assertTrue(rt.matchesMQTT("abcdxyz"));
-        assertFalse(rt.matchesMQTT("abc123xyz456/89"));
+        rt.add("nowildcard");
+        assertTrue(rt.matchesStandard("nowildcard"));
+        assertFalse(rt.matchesStandard("topic"));
+
+        assertTrue(rt.matchesMQTT("nowildcard"));
+        assertFalse(rt.matchesMQTT("topic"));
+
+        // Test wildcards in middle
+        rt.add("abc*xy*z");
+        assertTrue(rt.matchesStandard("abc123xyabc!@/<>#$%^&*()_+-=z" ));
```

Comment:
This input contains *. What would the resource string look like to allow that? Does IoT Core allow it? Add test case for it

Reply:
The resource one above this line will allow it `rt.add("abc*xy*z")`.

Reply:
I remember secret arn contains some special characters like _  Those are covered in this case right?

Reply:
Documenting another offline discussion with Vaibhav: IAM would escape this with `${*}`. There's also the `?` wildcard for matching a single character

https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_variables.html#policy-vars-infotouse

I'd also like to see that same behavior here. We can do this incrementally, but at a minimum we should require `?` and `$` to be escaped

Note: this would be backward incompatible. Need to think about this would impact customers who might be subscribing to `$aws` topics